src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -0,0 +1,581 @@
+        // Throws if any conversion from string to the expected data type fails.
+        internal void ReplaceNormalAttributesWithExtended(IEnumerable<KeyValuePair<string, string>> extendedAttributesEnumerable)
+        {
+            Dictionary<string, string> ea = new Dictionary<string, string>(extendedAttributesEnumerable);
+            if (ea.Count == 0)
+            {
+                return;
+            }
+            _extendedAttributes ??= new Dictionary<string, string>();
+
+            // First step: Insert or replace all the elements in the passed dictionary into the current header's dictionary.
+            foreach ((string key, string value) in ea)
+            {
+                _extendedAttributes[key] = value;
+            }
+
+            // Second, find all the extended attributes with known names and save them in the expected standard attribute.
+            if (ea.TryGetValue(PaxEaName, out string? paxEaName))
+            {
+                _name = paxEaName;
+            }
+            if (ea.TryGetValue(PaxEaLinkName, out string? paxEaLinkName))
+            {
+                _linkName = paxEaLinkName;
+            }
+            if (ea.TryGetValue(PaxEaATime, out string? paxEaATime))
+            {
+                if (TarHelpers.TryConvertToDateTimeOffset(paxEaATime, out DateTimeOffset aTime))
+                {
+                    _aTime = aTime;
+                }
+            }
+            if (ea.TryGetValue(PaxEaCTime, out string? paxEaCTime))
+            {
+                if (TarHelpers.TryConvertToDateTimeOffset(paxEaCTime, out DateTimeOffset cTime))
+                {
+                    _cTime = cTime;
+                }
+            }
+            if (ea.TryGetValue(PaxEaMTime, out string? paxEaMTime))
+            {
+                if (TarHelpers.TryConvertToDateTimeOffset(paxEaMTime, out DateTimeOffset mTime))
+                {
+                    _mTime = mTime;
+                }
+            }
+            if (ea.TryGetValue(PaxEaMode, out string? paxEaMode))
+            {
+                _mode = Convert.ToInt32(paxEaMode);
+            }
+            if (ea.TryGetValue(PaxEaSize, out string? paxEaSize))
+            {
+                _size = Convert.ToInt32(paxEaSize);
+            }
+            if (ea.TryGetValue(PaxEaUid, out string? paxEaUid))
+            {
+                _uid = Convert.ToInt32(paxEaUid);
+            }
+            if (ea.TryGetValue(PaxEaGid, out string? paxEaGid))
+            {
+                _gid = Convert.ToInt32(paxEaGid);
+            }
+            if (ea.TryGetValue(PaxEaUName, out string? paxEaUName))
+            {
+                _uName = paxEaUName;
+            }
+            if (ea.TryGetValue(PaxEaGName, out string? paxEaGName))
+            {
+                _gName = paxEaGName;
+            }
+            if (ea.TryGetValue(PaxEaDevMajor, out string? paxEaDevMajor))
+            {
+                _devMajor = int.Parse(paxEaDevMajor);
```

Comment:
@stephentoub do we have the ability to enable CA1305 just for this library? I'm not sure whether we run that flavor of analyzer.

Reply:
I´m hoping my last commit fixes the culture CI failures, but I also added this suggestion to the issue I created for the follow-up tasks: https://github.com/dotnet/runtime/issues/68230

Reply:
> I'm not sure whether we have the ability to add CA1305

Turning on CA1305 for the repo and doing `build clr.corelib+libs -rc release -warnaserror 0` produces 6,429 warnings (and that's one configuration only). I sanity-checked a few, and all that I looked at were false positives.

> do we have the ability to enable CA1305 just for this library?

Yes, if you _really_ want to, you can create a `.editorconfig` file in the src directory for the library and put this in it:
```
[*.cs]

# CA1305: Specify IFormatProvider
dotnet_diagnostic.CA1305.severity = warning
```

Reply:
What kind of false positives -- they were formatting for display? The rule AFAIK flags when we do not pass a format provider, so (I think) that can be universally applied - it might simply be redundant in some cases.

Reply:
If the analyzer flagged a case that is meant for display, then the call-site should pass in `CultureInfo.CurrentCulture`/`.CurrentUICulture`. The analyzer doesn't care *what* format provider you pass in, just that you're explicit about what you want, even if that's matching the otherwise implicit behavior.

Reply:
> then the call-site should pass in CultureInfo.CurrentCulture/.CurrentUICulture

That adds unnecessary expense.  For example, if you call Int32.ToString() and the value is non-negative, it will never have to access CultureInfo.CurrentInfo; if the call site always passes in CurrentCulture, you're now paying for that access regardless.

It also adds unnecessary verbosity... that's 6,429 places extra verbiage is required.

> What kind of false positives

Formatting for display, passed in a case where it won't actually be used, etc.

It also has tons of false negatives, e.g. it won't flag places ToString() is used implicitly, such as in string interpolation, as part of string concatenation, and so on.

> it might simply be redundant in some cases.

In every case I looked at.